lib/mayaUsd/render/vp2RenderDelegate/mayaPrimCommon.cpp

```diff
@@ -806,10 +806,15 @@ void MayaUsdRPrim::_SyncDisplayLayerModesInstanced(SdfPath const& id, unsigned i
 
     _forcedReprFlags = 0;
     _requiredModFlagsBitset.reset();
-    if (_useInstancedDisplayLayerModes) {
+    if (_useInstancedDisplayLayerModes && instanceCount > 0) {
         _displayLayerModesInstanced.resize(instanceCount);
+        InstancePrimPaths& instancePrimPaths = MayaUsdCustomData::GetInstancePrimPaths(id);
```

Comment:
I'm not sure if using the cached MayaUsdCustomData here is valid and correct. That is why I opened this code review to validate with someone more familiar with the code than I.

Reply:
From what I see in the code, this cache is updated inside mesh.cpp, in the cycle below line 1995. And you use this cache in the function _SyncDisplayLayerModesInstanced just above this line. So it seems like you use the cache before it's updated. Apart from that, using this cache seems like a good idea. Although, to calculate the cache, we still need to call GetScenePrimPath functions

Reply:
Ah, that would explain the failed tests! Thank you.